### PR TITLE
test: add unit tests for alias/tags DB functions

### PR DESCRIPTION
**Pull request by lmphil**
## Description

This change is meant to add unit tests to DB functions related to the "alias" and "tags" features.

## Type of change

- [x] Unit Tests

## How Has This Been Tested?

- [x] Locally

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: internal/pkg/db/db_test.go [hunk 1]
```diff
@@ -62,6 +65,16 @@ func (dbs *dbSuite) SetupSuite() {
 			dbs.FailNow("failed storing document", "err", err)
 		}
 
+		name := strings.Split(sboms[sbomIdx].Name(), ".")[1]
+		if err := backend.SetUniqueAnnotation(doc.GetMetadata().GetId(), db.AliasAnnotation, name); err != nil {
+			dbs.FailNow("failed to set alias", "err", err)
+		}
```
Comment (jhoward-lm) [inline internal/pkg/db/db_test.go:71]:
```suggestion
		dbs.Require().NoError(
			backend.SetUniqueAnnotation(doc.GetMetadata().GetId(), db.AliasAnnotation, name), 
			"failed to set alias", "err", err,
		)```

File: internal/pkg/db/db_test.go [hunk 2]
```diff
@@ -62,6 +65,16 @@ func (dbs *dbSuite) SetupSuite() {
...
+
+		if err := backend.AddAnnotations(doc.GetMetadata().GetId(),
+			db.TagAnnotation, documentTags[sbomIdx]...); err != nil {
+			dbs.FailNow("failed to add tags", "err", err)
+		}
```
Comment (jhoward-lm) [inline internal/pkg/db/db_test.go:76]:
```suggestion
		dbs.Require().NoError(
			backend.AddAnnotations(doc.GetMetadata().GetId(), db.TagAnnotation, documentTags[sbomIdx]...), 
			"failed to add tags", "err", err,
		)
```

File: internal/pkg/db/db_test.go [hunk 3]
```diff
@@ -93,6 +106,114 @@ func (dbs *dbSuite) TestGetDocumentByID() {
...
+	dbs.Require().NoError(err)
+	dbs.Require().Empty(filteredDocs5)
+
+	// No tags
+	filteredDocs6, err := dbs.backend.FilterDocumentsByTag(docs)
+	dbs.Require().NoError(err)
+	dbs.Require().Len(filteredDocs6, 2)
+	dbs.Require().Equal(filteredDocs6[0].GetMetadata().GetId(), dbs.documents[0].GetMetadata().GetId())
+	dbs.Require().Equal(filteredDocs6[1].GetMetadata().GetId(), dbs.documents[1].GetMetadata().GetId())
+}
```
Comment (jhoward-lm) [inline internal/pkg/db/db_test.go:181]: This should be a loop and create a subtest for each test case.

For example:

```go
	for _, data := range []struct {
		name     string
		tags     []string
		expected []*sbom.Document
	}{
		{
			name:     "tag1",
			tags:     []string{"tag1"},
			expected: []*sbom.Document{dbs.documents[0]},
		},
	} {
		dbs.Run(data.name, func() {
			filteredDocs, err := dbs.backend.FilterDocumentsByTag(docs, data.tags...)
			dbs.Require().NoError(err)
			dbs.Require().Len(filteredDocs, len(data.expected))

			for idx := range data.expected {
				dbs.Require().Equal(filteredDocs[idx].GetMetadata().GetId(), docs[idx].GetMetadata().GetId())
			}
		})
	}
```

File: internal/pkg/db/db_test.go [hunk 4]
```diff
@@ -93,6 +106,114 @@ func (dbs *dbSuite) TestGetDocumentByID() {
...
+
+func (dbs *dbSuite) TestSetAlias() {
```
Comment (jhoward-lm) [inline internal/pkg/db/db_test.go:183]: This should be a loop and create a subtest for each test case. See previous example

**Comment by jhoward-lm** [comment]
PR title must be conventional commit and should use standard casing. 

e.g. `test: add unit tests for alias/tags DB functions`

File: internal/pkg/db/db_test.go [hunk 5]
```diff
@@ -38,6 +39,8 @@ type dbSuite struct {
 	documents []*sbom.Document
 }
 
+var documentTags = [][]string{{"tag1", "tag2"}, {"tag2", "tag3"}}
+
 func (dbs *dbSuite) SetupSuite() {
 	backend, err := db.NewBackend(db.WithDatabaseFile(db.DatabaseFile))
```
Comment (jhoward-lm) [inline internal/pkg/db/db_test.go:45]: Test using in-memory database if possible

```suggestion
	backend, err := db.NewBackend(db.WithDatabaseFile(":memory:"))
```

File: internal/pkg/db/db_test.go [hunk 6]
```diff
@@ -62,6 +65,16 @@ func (dbs *dbSuite) SetupSuite() {
 			dbs.FailNow("failed storing document", "err", err)
 		}
 
+		name := strings.Split(sboms[sbomIdx].Name(), ".")[1]
```
Comment (jhoward-lm) [inline internal/pkg/db/db_test.go:68]: Can you change `sbomIdx` to just `idx`

File: internal/pkg/db/db_test.go [hunk 7]
```diff
@@ -93,6 +106,114 @@ func (dbs *dbSuite) TestGetDocumentByID() {
...
+
+	// Set Alias (Force)
+	err = dbs.backend.SetAlias(docs[0].GetMetadata().GetId(), "cdx2", true)
+	dbs.Require().NoError(err)
+	docAlias, err = dbs.backend.GetDocumentUniqueAnnotation(docs[0].GetMetadata().GetId(), db.AliasAnnotation)
+	dbs.Require().NoError(err)
+	dbs.Require().Equal("cdx2", docAlias)
+}
+
 func TestStoreSuite(t *testing.T) {
```
Comment (jhoward-lm) [inline internal/pkg/db/db_test.go:217]: Forgot to update this name when originally written

```suggestion
func TestDBSuite(t *testing.T) {
```

**Review by jhoward-lm — DISMISSED** [review]

**Comment by jhoward-lm** [comment]
Can you try updating `db.FilterDocumentsByTag` to leverage the `sliceutil` helper functions?

```go
func (backend *Backend) FilterDocumentsByTag(documents []*sbom.Document, tags ...string) ([]*sbom.Document, error) {
	taggedDocuments, err := backend.GetDocumentsByAnnotation(TagAnnotation, tags...)
	if err != nil {
		return nil, fmt.Errorf("failed to get documents by tag: %w", err)
	}

	taggedDocumentIDs := sliceutil.Extract(taggedDocuments, func(doc *sbom.Document) string {
		return doc.GetMetadata().GetId()
	})

	documents = sliceutil.Filter(documents, func(doc *sbom.Document) bool {
		return slices.Contains(taggedDocumentIDs, doc.GetMetadata().GetId())
	})

	return documents, nil
}
```

**Review by ashearin — APPROVED** [review]

**Pull request merged**